modules/standard/IO.chpl

```diff
@@ -7018,6 +7043,105 @@ iter fileReader.lines(stripNewline = false) {
   this.unlock();
 }
 
+@chpldoc.nodoc
+iter fileReader.lines(
+  param tag: iterKind,
+  stripNewline = false,
+  targetLocales: [?tld] locale = [this._home,]
+): string
+  where tag == iterKind.standalone
+{
+  on this._home {
+    try! this.lock();
+
+    var f = chpl_fileFromReaderOrWriter(this);
+
+    const myStart = qio_channel_get_start_pos(this._channel_internal),
+          myEnd = qio_channel_get_end_pos(this._channel_internal),
+          myBounds = myStart..<(try! min(myEnd, f.size));
+
+    if targetLocales.size == 1 && targetLocales.first == this._home {
+      // single locale case on this fileReader's home locale
+      const nTasks = here.maxTaskPar,
+            byteOffsets = try! findFileChunks(f, nTasks, myBounds);
```

Comment:
I have lost track. Do we not allow throwing from standalone iterators? If these `try!` are all a workaround for that, we need the documentation for this function to say that certain error cases currently `halt` when they will throw in the future.

Reply:
Yeah I believe that's the case. For the halts that are directly I/O related, I linked to the I/O docs section about that.

The other thing wrapped in `try!` were the `findFileChunks` calls, where it throws when the file cannot be split into the requested number of chunks. I've modified the code to fall back to serial iteration whenever that happens.